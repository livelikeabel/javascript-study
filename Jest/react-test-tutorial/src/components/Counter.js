import React, {Component} from 'react';

class Counter extends Component {
    state = {
        value: 1
    }
    onIncrease = () => {
        this.setState(({value}) => ({value: value + 1}));
    }
    onDecrease = () => {
        this.setState((state) => {return {value: state.value - 1}})
    }
    render() {
        return (
            <div>
                <h1>카운터</h1>
                <h2>{this.state.value}</h2>
                <button onClick={this.onIncrease}>+</button>
                <button onClick={this.onDecrease}>-</button>
            </div>
        )
    }
}

export default Counter;