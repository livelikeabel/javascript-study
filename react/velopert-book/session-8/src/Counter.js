import React, { useState } from 'react';

const Counter = () => {
  const [value, setValue] = useState(0);

  return (
    <div>
      <p>
        Current counter value is <b>{value}</b>
      </p>
      <button onClick={() => setValue(value + 1)}>+1</button>
      <button onClick={() => setValue(value - 1)}>-1</button>
    </div>
  );
};

export default Counter;
