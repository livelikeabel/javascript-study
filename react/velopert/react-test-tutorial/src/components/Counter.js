import React, { Component } from 'react'

class Counter extends Component {
  state = {
    value: 1
  }
  onIncrease = () => {
    this.setState(({ value }) => ({ value: value + 1 }))
  }
  onDecrease = () => {
    this.setState(({ value }) => ({ value: value - 1 }))
  }

  render() {
    return (
      <div>
        <h1>Counter</h1>
        <h2>{this.state.value}</h2>
        <button onClick={this.onIncrease}>+</button>
        <button onClick={this.onDecrease}>-</button>
      </div>
    )
  }
}

export default Counter
