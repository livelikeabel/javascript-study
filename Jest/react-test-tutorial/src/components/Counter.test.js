import React from 'react'
import renderer from 'react-test-renderer';
import Counter from './Counter';

describe('Counter', () => {
    let component = null;
    
    it('renders correctly', () => {
        component = renderer.create(<Counter />);
    });
    
    it('matches snapshot', () => {
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})