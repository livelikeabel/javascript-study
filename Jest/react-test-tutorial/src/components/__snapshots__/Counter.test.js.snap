// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter decreases correctly 1`] = `
<div>
  <h1>
    운터
  </h1>
  <h2>
    1
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;

exports[`Counter increases correctly 1`] = `
<div>
  <h1>
    운터
  </h1>
  <h2>
    2
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;

exports[`Counter matches snapshot 1`] = `
<div>
  <h1>
    운터
  </h1>
  <h2>
    1
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</div>
`;
